### ISSUE TITLE
(Godot 4.0.3 rc1) Navigation Agent weird behavior spiraling out of control, can't move to world origin?

### ISSUE THREAD

**Issue by NanotechGamedev** [godotengine/godot#77106]
### Godot version

v4.0.3.rc1.official [2d74ee0e5]

### System information

Windows 10, NVIDIA GeForce GTX 1050 Ti

### Issue description

Seems like due to the other units, the moving unit can't move to the world origin for some reason? See gif and project mrp, very weird behavior, thought it might be useful to find other bugs.

![NavigationIssue](https://github.com/godotengine/godot/assets/133692370/2da39c6a-5248-455f-8424-24d10cefca07)

Only node with a script is the moving unit,

### Steps to reproduce

Check project mrp and run the scene Debug with F6, left click to move the unit, moving the unit to the world origin will make it move everywhere around the origin, the cube mesh only represents the world origin,

### Minimal reproduction project

[nav_issue.zip](https://github.com/godotengine/godot/files/11480591/nav_issue.zip)

**Comment by smix8** [comment]
As you can see from the red pathfinding debug the path is calculated correct.

You are calculating your velocity wrong.

You can find examples scripts to use a NavigationAgent with a CharacterBody3D in the navigation documentation [here](https://docs.godotengine.org/en/latest/tutorials/navigation/navigation_using_navigationagents.html#actor-as-characterbody3d).

**Comment by tavurth** [comment]
Workaround for the above issue if confirmed:

```gdscript
if abs(target.x) < 0.001 and abs(target.y) < 0.001 and abs(target.z) < 0.0001:
  return false
```

**Comment by smix8** [comment]
What causes the problem around the world origin is 2 invisible avoidance agents still standing there cause they never receive their first position update. If the RigidBody3D or Character3D are moved so they update their position the first time (or removed from the scene) the problem goes away.

This should be already fixed in recent master / Godot 4.1 cause there it sets the avoidance position when a NavigationAgent node enters the SceneTree but this was not the case in Godot 4.0.

**Comment by smix8** [comment]
Even with the linked fix or suggested workaround in Godot 4.0.x the agent radius should properly be set sightly larger for a decent result in this test project.

Godot 4.0.x uses the RVO 3D avoidance for a pseudo 2D avoidance when `ignore_y` is enabled. The issue with that becomes very apparent when `ignore_y` is disabled on the NavigationAgent3D. The agent, when very close to other agents or obstacles, tries to avoid upward or downward like it would in the unchained 3D avoidance but can't so it only moves with the tiny xz-axis velocity that is left. Having a larger radius, while not ideal, at least makes it avoid to the sides most of the time.

**Comment by akien-mga** [comment]
Fixed by #77110.
